Yosemite/YosemiteTests/Stores/TaxStoreTests.swift

```diff
@@ -191,12 +191,28 @@ final class TaxClassStoreTests: XCTestCase {
         XCTAssertEqual(taxClass1?.toReadOnly(), sampleTaxClassMutated())
         XCTAssertEqual(viewStorage.countObjects(ofType: Storage.TaxClass.self), 1)
     }
+
+    func test_retrieveTaxRates_then_persists_TaxRates() {
+        network.simulateResponse(requestUrlSuffix: "taxes", filename: "taxes")
+        XCTAssertEqual(viewStorage.countObjects(ofType: Storage.TaxRate.self), 0)
+
+        // When
+        let result: Result<[Yosemite.TaxRate], Error> = waitFor { promise in
+            let action = TaxAction.retrieveTaxRates(siteID: self.sampleSiteID, pageNumber: 1, pageSize: 25) { result in
+                promise(result)
+            }
+            self.store.onAction(action)
+        }
```

Comment:
This is more a question than a suggestion, should we care about _weakifying_ self here? Since the test is very short lived, even if there's a retain cycle everything will be reset on tear-down. I don't see it as a concern, just as sort of consistency. 
```suggestion
        let result: Result<[Yosemite.TaxRate], Error> = waitFor { [weak self] promise in
            guard let self = self else { return }
            let action = TaxAction.retrieveTaxRates(siteID: self.sampleSiteID, pageNumber: 1, pageSize: 25) { result in
                promise(result)
            }
            self.store.onAction(action)
        }
```

Reply:
That makes sense to make it consistent, done in 00899e0d34a88251c99d35d980e8ebef585697f9